compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -17762,15 +17762,29 @@ TR::Node *fmaxminSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
          fmin = fmax = secondChild->getFloat();
       else
          {
-         if (firstChild->getFloat() <= secondChild->getFloat())
+         float first = firstChild->getFloat();
+         float second = secondChild->getFloat();
+         int32_t firstBits = *reinterpret_cast<int32_t *>(&first);
+         int32_t secondBits = *reinterpret_cast<int32_t *>(&second);
+         int32_t nZeroBits = 0x80000000;
+         int32_t pZeroBits = 0;
+
+         // compare int32_t bits to avoid compiler simplifying zeros (just in case it treats -0.0 literal as +0.0)
+         if ((firstBits == pZeroBits || firstBits == nZeroBits) && (secondBits == pZeroBits || secondBits == nZeroBits))
+            {
+            int32_t sign = firstBits & nZeroBits;
+            fmin = sign < 0 ? first : second;
```

Comment:
Couple of thoughts here, do we really need to know sign here? Also I feel we can study the cases and can achieve what we want to in two conditional block only (We can try to avoid if/else if/else if possible).